### PR TITLE
Fix raytracing and descriptor indexing crashes

### PR DESCRIPTION
**Pull request by jeffbolznv**
This is WIP because the raytracing fix is in https://github.com/KhronosGroup/SPIRV-Tools/pull/2190, but that needs to get merged and pulled into glslang and then into this PR.

### CODE REVIEW

File: layers/descriptor_sets.cpp [hunk 1]
```diff
@@ -1543,7 +1543,9 @@ void cvdescriptorset::ImageSamplerDescriptor::UpdateDrawState(layer_data *dev_da
     if (iv_state) {
         core_validation::AddCommandBufferBindingImageView(dev_data, cb_node, iv_state);
     }
-    SetImageViewLayout(dev_data, cb_node, image_view_, image_layout_);
+    if (image_view_) {
+        SetImageViewLayout(dev_data, cb_node, image_view_, image_layout_);
```
Comment (jeffbolznv) [inline layers/descriptor_sets.cpp]: This crashed in an app using descriptor_indexing with some unbound descriptors.

**Comment by jeffbolznv** [comment]
Removed WIP. This now includes the glslang/spirv-tools update that fixes the main bug, and is ready to be reviewed/merged.

**Comment by jeffbolznv** [comment]
Can this be merged please? All feedback has been addressed.

**Comment by tobine** [comment]
Yep, LGTM, merged. Thanks @jeffbolznv and sorry for delay!

**Pull request merged**